src/MoBi.Core/Services/BuildingBlockVersionUpdater.cs

```diff
@@ -29,15 +29,19 @@ public BuildingBlockVersionUpdater(IMoBiProjectRetriever projectRetriever, IEven
          _dialogCreator = dialogCreator;
       }
 
-      public void UpdateBuildingBlockVersion(IBuildingBlock buildingBlock, uint newVersion)
+      public virtual void UpdateBuildingBlockVersion(IBuildingBlock buildingBlock, uint newVersion, bool shouldConvertPKSimModule)
       {
          if (buildingBlock == null)
             return;
 
-         if (shouldShowModuleConversionWarning(buildingBlock))
+         if (shouldConvertPKSimModule && shouldShowModuleConversionWarning(buildingBlock))
```

Comment:
I would move this bool check into the `shouldShow` method. Then I would take line 43 and move it inside this condition. The problem with line 42/43 is that not all building blocks belong to a Module. This method `shouldShow` also checks for that.

Basically, the conditions for showing the warning are the same for the conditions to convert the module, so let's only check it once.